docs/admin/deployment.md

```diff
@@ -216,24 +223,29 @@ Each generated engine section will specify a SCM storage tier (PMem or tmpfs) in
 more NVMe storage tiers. All hardware components specified in an engine config section should be
 bound to the same NUMA node (PMem bdev, SSDs and host fabric interface).
 
-- `--scm-only` request that a config without NVMe should be generated. This flag will override the
+- `--scm-only` requests that a config without NVMe should be generated. This flag will override the
 command's normal behavior and should be used only in circumstances where NVMe SSDs are unavailable
 or not balanced across NUMA nodes and multiple engines are required per host. Note that DAOS
 performance will be suboptimal without NVMe SSDs.
 
 - `--net-class` selects a specific network device class, options are `ethernet` or `infiniband`.
 If not set explicitly on the commandline, default is `infiniband`. This option will alter the
 command's behavior and should only be used when normal command operation is not sufficient.
-Note that DAOS performance may be suboptimal if ethernet devices are used instead of infiniband.
 
 - `--net-provider` selects a specific network provider , this can be set to any provider string
 supported on the hosts e.g. ofi+tcp. This option will alter the command's behavior and should only
 be used when normal command operation is not sufficient. Note that specifying a provider will
 prevent the command from selecting the best available.
 
 - `--use-tmpfs-scm` will produce a config specifying RAM-disk (tmpfs) devices in the first (SCM)
-storage tier. The RAM-disk sizes will be calculated based on using 75% of the host's total
-memory (as reported by `/proc/meminfo`).
+storage tier. The RAM-disk sizes will be calculated based on the host's total memory (as reported
+by `/proc/meminfo`). An MD-on-SSD config will be generated if this flag is set.
```

Comment:
Why does this flag enable MD on SSD? I would think that should be keyed off of the `--control-metadata-path` flag instead, as that's what the server logic does.

Reply:
I could make that change, currently the behaviour is as described. Would you like to do that in this PR?

Reply:
@kjacque: Is my understanding correct re: the server logic here? If we've got an inconsistency then it seems like it should be resolved...

Reply:
what's the inconsistency? the config generate flag is explicit and indicates that the user wants to generate a tmpfs config rather than use any present pmem. IMO that's valid and in this case explicit is better than implicit. I'm happy to change if the consensus is strong that we should automate that selection based on the presence of another option but I'm not seeing any obvious inconsistency.

Reply:
> what's the inconsistency?

If I am understanding correctly, based on the documentation here, using the `--use-tmpfs-scm` flag enables MD-on-SSD mode. Whereas when the configuration is generated manually, MD-on-SSD mode is only enabled when the `control_metadata:` parameter is set. I'm just saying that it's inconsistent (and confusing) to have two different ways of enabling MD-on-SSD. Furthermore, doing it this way would mean that there's no way to use tmpfs for emulated pmem in non-MD-on-SSD mode anymore, at least via generated configuration.

Reply:
ok, so I will remove the --use-tmpfs-scm flag and enable tmpfs-MD-on-SSD config generation when control_metadata commandline option is supplied. does that sound like what you want?

Reply:
I'm not objecting to the `--use-tmpfs-scm` flag itself. I think there's still a use case for emulating pmem in non-MD-on-SSD (aka legacy) mode. I just wanted to point out that the code as implemented currently uses the presence of a populated `control_metadata:` section in the server config to toggle between the modes. So it didn't make sense to me to tie the mode toggle to the use of tmpfs or not.

Reply:
I will update to allow for ephemeral config and key MD-on-SSD off control_metadata option.

Reply:
done, please could you review @mjmac